internal/chain/reward.go

```diff
@@ -144,6 +168,19 @@ func AccumulateRewards(
 	if bc.Config().IsStaking(header.Epoch()) &&
 		bc.CurrentHeader().ShardID() == shard.BeaconChainShardID {
 
+		defaultReward := BlockRewardStakedCase
+
+		if shard.Schedule.IsLastBlock(blockNum) {
```

Comment:
Should use ShardState size > 0 for the condition of last block. The schedule is only for pre-staking.